homeassistant/components/pvoutput/sensor.py

```diff
@@ -78,62 +70,30 @@ class PvoutputSensor(SensorEntity):
     _attr_device_class = SensorDeviceClass.ENERGY
     _attr_native_unit_of_measurement = ENERGY_WATT_HOUR
 
-    def __init__(self, rest, name):
+    def __init__(self, pvoutput, status, name):
         """Initialize a PVOutput sensor."""
-        self.rest = rest
         self._attr_name = name
-        self.pvcoutput = None
-        self.status = namedtuple(
-            "status",
-            [
-                ATTR_DATE,
-                ATTR_TIME,
-                ATTR_ENERGY_GENERATION,
-                ATTR_POWER_GENERATION,
-                ATTR_ENERGY_CONSUMPTION,
-                ATTR_POWER_CONSUMPTION,
-                ATTR_EFFICIENCY,
-                ATTR_TEMPERATURE,
-                ATTR_VOLTAGE,
-            ],
-        )
+        self.pvoutput = pvoutput
+        self.status = status
 
     @property
     def native_value(self):
         """Return the state of the device."""
-        if self.pvcoutput is not None:
-            return self.pvcoutput.energy_generation
-        return None
+        return self.status.energy_generation
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes of the monitored installation."""
-        if self.pvcoutput is not None:
-            return {
-                ATTR_ENERGY_GENERATION: self.pvcoutput.energy_generation,
-                ATTR_POWER_GENERATION: self.pvcoutput.power_generation,
-                ATTR_ENERGY_CONSUMPTION: self.pvcoutput.energy_consumption,
-                ATTR_POWER_CONSUMPTION: self.pvcoutput.power_consumption,
-                ATTR_EFFICIENCY: self.pvcoutput.efficiency,
-                ATTR_TEMPERATURE: self.pvcoutput.temperature,
-                ATTR_VOLTAGE: self.pvcoutput.voltage,
-            }
+        return {
+            ATTR_ENERGY_GENERATION: self.status.energy_generation,
+            ATTR_POWER_GENERATION: self.status.power_generation,
+            ATTR_ENERGY_CONSUMPTION: self.status.energy_consumption,
+            ATTR_POWER_CONSUMPTION: self.status.power_consumption,
+            ATTR_EFFICIENCY: self.status.normalized_ouput,
```

Comment:
Is that by intention - changing `pvcoutput.efficiency` to `status.normalized_ouput`?

Reply:
Yes. It is displayed as efficiency everywhere, but the API calls it differently.
For the library, I kept to the API spec.

Reply:
Noticed a typo in that variable now as well `ouput` -> `output`, thanks auto-complete 😄 

Will correct that in a next bump (will need a couple of those anyways).